src/main/java/org/snomed/snowstorm/core/data/services/AutoMerger.java

```diff
@@ -0,0 +1,542 @@
+
+                mergedReferenceSetMember.setReferencedComponentId(sourceReferenceSetMember.getReferencedComponentId());
+                mergedReferenceSetMember.setActive(getValueChanged(targetReferenceSetMemberOld.isActive(), targetReferenceSetMemberNew.isActive(), sourceReferenceSetMember.isActive()));
+                mergedReferenceSetMember.setReleased(getValueChanged(targetReferenceSetMemberOld.isReleased(), targetReferenceSetMemberNew.isReleased(), sourceReferenceSetMember.isReleased()));
+                mergedReferenceSetMember.setRefsetId(getValueChanged(targetReferenceSetMemberOld.getRefsetId(), targetReferenceSetMemberNew.getRefsetId(), sourceReferenceSetMember.getRefsetId()));
+                mergedReferenceSetMember.setModuleId(getValueChanged(targetReferenceSetMemberOld.getModuleId(), targetReferenceSetMemberNew.getModuleId(), sourceReferenceSetMember.getModuleId()));
+                mergedReferenceSetMember.setAdditionalField(
+                        ReferenceSetMember.OwlExpressionFields.OWL_EXPRESSION,
+                        getValueChanged(
+                                targetReferenceSetMemberOld.getAdditionalField(ReferenceSetMember.OwlExpressionFields.OWL_EXPRESSION),
+                                targetReferenceSetMemberNew.getAdditionalField(ReferenceSetMember.OwlExpressionFields.OWL_EXPRESSION),
+                                sourceReferenceSetMember.getAdditionalField(ReferenceSetMember.OwlExpressionFields.OWL_EXPRESSION)
+                        )
+                );
+            }
+
+            mergedReferenceSetMember.updateEffectiveTime();
+            // If changed, ensure component is re-activated unless the RHS had the inactivation.
+            if (mergedReferenceSetMember.getEffectiveTime() == null && sourceAxiom != null && sourceAxiom.getReferenceSetMember() != null && !sourceAxiom.getReferenceSetMember().isActive() && targetAxiomNew.isActive()) {
+                mergedReferenceSetMember.setActive(true);
+                mergedAxiom.setActive(true);
+            }
+            mergedAxiom.setReferenceSetMember(mergedReferenceSetMember);
+            mergedAxioms.put(changedAxiomId, mergedAxiom);
+        }
+
+        // Merge unchanged Axioms
+        for (Map.Entry<String, Axiom> entrySet : sourceAxioms.entrySet()) {
+            String unchangedAxiomId = entrySet.getKey();
+            Axiom axiom = entrySet.getValue();
+
+            // There are no target edits if the identifier is absent; therefore, take from source.
+            mergedAxioms.putIfAbsent(unchangedAxiomId, axiom);
+        }
+
+        return new HashSet<>(mergedAxioms.values());
+    }
+
+    private void reapplyLangMemberChanges(Map<String, SnomedComponent<?>> sourceLangMembers, Map<String, SnomedComponent<?>> targetLangMembersOld, Map<String, SnomedComponent<?>> targetLangMembersNew, Set<String> changedLangMemberIds, Concept mergedConcept) {
```

Comment:
SnomedComponent<?> should be ReferenceSetMember type

Reply:
I have made this change.